src/SFA.DAS.EAS.Portal/Application/Commands/Cohort/CohortApprovalRequestedCommandHandler.cs

```diff
@@ -0,0 +1,83 @@
+﻿using SFA.DAS.Commitments.Api.Client.Interfaces;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Linq;
+using System;
+using SFA.DAS.EAS.Portal.Application.Services;
+using SFA.DAS.EAS.Portal.Client.Types;
+using SFA.DAS.EAS.Portal.Application.Commands.Account;
+using SFA.DAS.HashingService;
+
+namespace SFA.DAS.EAS.Portal.Application.Commands.Cohort
+{
+    public class CohortApprovalRequestedCommandHandler : ICommandHandler<CohortApprovalRequestedCommand>
+    {
+        private readonly IAccountDocumentService _accountsService;
+        private readonly IProviderCommitmentsApi _providerCommitmentsApi;
+        private readonly ICommandHandler<AccountCreatedCommand> _accountCreatedHandler;
+        private readonly IHashingService _hashingService;
+
+        public CohortApprovalRequestedCommandHandler(
+            IAccountDocumentService accountsService, 
+            IProviderCommitmentsApi providerCommitmentsApi,
+            ICommandHandler<AccountCreatedCommand> accountCreatedHandler,
+            IHashingService hashingService)
+        {
+            _accountsService = accountsService;
+            _providerCommitmentsApi = providerCommitmentsApi;
+            _accountCreatedHandler = accountCreatedHandler;
+            _hashingService = hashingService;
+        }
+
+        public async Task Handle(CohortApprovalRequestedCommand command, CancellationToken cancellationToken = default)
+        {
+            var commitment = await _providerCommitmentsApi.GetProviderCommitment(command.ProviderId, command.CommitmentId);            
+            long accountLegalEntityId = _hashingService.DecodeValue(commitment.AccountLegalEntityPublicHashedId);
+
+            await _accountCreatedHandler.Handle(new AccountCreatedCommand(command.AccountId, commitment.LegalEntityName), cancellationToken);
+
+            var accountDocument = await _accountsService.Get(command.AccountId, cancellationToken);           
+            var account = accountDocument.Account;
+            var cohortReference = commitment.Reference;
+
+            var organisation = account.Organisations.FirstOrDefault(o => o.AccountLegalEntityId.Equals(accountLegalEntityId));
+            if (organisation == null)
+            {
+                organisation = new Organisation()
+                {
+                    AccountLegalEntityId = accountLegalEntityId,
+                    Name = commitment.LegalEntityName
+                };
+
+                account.Organisations.Add(organisation);
+            }
+
+            var cohort = organisation.Cohorts.FirstOrDefault(c => c.Id != null && c.Id.Equals(cohortReference, StringComparison.OrdinalIgnoreCase));
+
+            if (cohort == null)
+            {
+                cohort = new Client.Types.Cohort { Id = cohortReference };
```

Comment:
We ignore the Id in the `CommitmentView` returned from the commitments API call, and instead store the Reference, but it's stored locally as the Cohort's Id.

What's the rational behind this? Can we not store the commitment id? If it does make sense to store the reference, rather than the Id, wouldn't it be better to keep the name Reference to reduce confusion?

Reply:
We now store the commitment id, although we should be storing it as it's native type of long, rather than converting it to a string. Also, we're still storing the Reference. Not sure if there is any need.